account/lnurl.go

```diff
@@ -29,14 +30,60 @@ type LoginResponse struct {
 	Token string `json:"token"`
 }
 
-func (a *Service) HandleLNURL(rawString string) (*data.LNUrlResponse, error) {
+func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err error) {
+	var iparams lnurl.LNURLParams
+	var rawurl, lightningAddress string
+
 	encodedLnurl, ok := lnurl.FindLNURLInText(rawString)
-	if !ok {
-		return nil, fmt.Errorf("'%s' does not contain an LNURL.", rawString)
+	if ok {
+		a.log.Infof("HandleLNURL %v", encodedLnurl)
+		rawurl, iparams, err = lnurl.HandleLNURL(encodedLnurl)
+
+	} else {
+
+		err = fmt.Errorf("%q does not contain an LNURL.", rawString)
+
+		if strings.Contains(rawString, "@") {
+			// ref. Lightning-Address https://github.com/andrerfneves/lightning-address/blob/master/DIY.md
+
+			err = nil
+			parts := strings.Split(rawString, "@")
+			if len(parts) != 2 {
+				err = fmt.Errorf("%q is not an e-mail address", rawString)
+			} else {
+
+				url := &url.URL{
+					Scheme: "https",
+					Host:   parts[1],
+					Path:   ".well-known/lnurlp/" + parts[0],
+				}
+
+				resp, err := http.Get(url.String())
```

Comment:
Did you try calling `lnurl. HandleLNURL` here? This function has the ability to handle 'https' prefixed urls and perform the get request.